tests/lean/679.lean.expected.out

```diff
@@ -1 +1 @@
-Except.ok Lean.Name.anonymous
+Except.ok ([anonymous])
```

Comment:
Something that's consistently confusing is whether the output of `#eval` should be valid code via `Repr`, or if it can be any string representation whatsoever via `ToString` too. In any case the output here is a bit weird because it is mixing the valid code `Except.ok (..)` representation with `ToString Name`. Both the previous version and `ok: [anonymous]` seem to make more sense. I thought the new `scoped instance (priority := low - 10) [ToString α] : Repr α where` was causing this but it still appears after removing the instance so I am not sure about the reason.

Reply:
Indeed, it's not completely what the output of `#eval` should be.  It used to take either a `ToString` or a `Repr` instance, so you already weren't guaranteed to get a "valid code" output.

The "culprit" here is the new instance:
```lean
scoped instance [ToMessageData ε] [ToMessageData α] : ToMessageData (Except ε α)
```
Which raises another related question: should `MessageData` be more like `Repr` or like `ToString`?

Reply:
As for what it *should* be: for me the most important part is to get *some* output, and that expressions are clickable.  None of the `ToMessageData`/`Repr`/`ToString` classes have instances for everything, so we'll need to mix them for some time.

Reply:
> Which raises another related question: should MessageData be more like Repr or like ToString?

I'm afraid the answer is that we want both. We use the class for many specific types like Expr and Syntax where it is like `ToString`, but then there are instance like for `List`, and `Except` above, that should definitely be `Repr`-like (which would also properly handle the parentheses).

Reply:
> that should definitely be Repr-like (which would also properly handle the parentheses).

Are you suggesting that `toMessageData` should have a precedence argument?

Reply:
I think we would a new, separate typeclass `ReprToMessageData` (or whatever), which should have a precedence argument, yeah